### ISSUE TITLE
Adding ability to print full namespace of module for CallsiteParameterAdder

### ISSUE THREAD

**Issue by macintacos** [hynek/structlog#492]
When using `CallsiteParameter.MODULE` in `CallsiteParameterAdder`, it's a bit too vague for my tastes - for example, a module named `config` has no meaning unless I know what package it's referring to. For example, I may have a module called `foo.bar.config` and another package could have a module called `baz.buz.config`, and it would not be possible to tell the difference with that value currently.

I know that there's `CallsiteParameter.PATHNAME`, but that's overly verbose - although I guess I could transform it to be something more palatable... At the end of the day though, it would be nice if there was a `CallsiteParameter` that would just print out `foo.bar.config`.

**Comment by hynek** [comment]
Those values and behavior are based on stdlib logging – how do they handle it?

**Comment by pahrohfit** [comment]
> 

> Those values and behavior are based on stdlib logging – how do they handle it?

poorly.

zero tolerance for deviation or extension. I have a patch for this (new `provider`), but waiting for my last pull because I'm sometimes lazy and potentially forgot to branch.

**Comment by pahrohfit** [comment]
I couldn't find an easy/transportable way to add it as a `CallsiteParameterAdder`, but I was able to toss it into a small processor #568

**Comment by decaz** [comment]
@hynek I'm struggling with the same problem of logging neat path to module and have custom processor which doesn't use frame. I suggest to consider the following handler of `CallsiteParameter.MODULE`:
```python
def _get_callsite_module(module: str, frame: FrameType) -> Any:
    return module
```
What do you think?

**Comment by hynek** [comment]
> [@hynek](https://github.com/hynek) I'm struggling with the same problem of logging neat path to module and have custom processor which doesn't use frame. I suggest to consider the following handler of `CallsiteParameter.MODULE`:
> 
> def _get_callsite_module(module: str, frame: FrameType) -> Any:
>     return module
> What do you think?

I'm sorry, but I'm not sure what you exactly mean. Please, be a bit more verbose with what you want to achieve and how that function is supposed to be used and be of help – thanks.

**Comment by decaz** [comment]
@hynek I mean logging of full path to the module, for instance: `foo.bar.baz`. Currently `CallsiteParameter.MODULE` will log just `baz` or `__init__` if logging was done within `foo.bar.__init__` module.

**Comment by hynek** [comment]
I believe this is fixed by `CallsiteParameter.QUAL_NAME`.

**Comment by vincent-stater** [comment]
This is actually not fixed by `QUAL_NAME`. `qual_name` gives you full "path" within a module. Say: "A.b.C.d" can point to:

```py
# foo/bar/baz.py
class A:
  def b(self):
    class C:
      def d(self):
        log.info("Hello!")
```

I believe this request is for the full "import path" to the module in question. For example: the above module would be importable using "foo.bar.baz".  However, `module` will provide only `baz` instead of the desired `foo.bar.baz`. And `qual_name` will still only look at the  location _within_ the module.

You can get this path using

```py
frame.f_globals.get("__name__", "__main__")
```` 

---

I'm actually looking for a similar feature to migrate away from `loguru`, where this is the standard format.

**Comment by decaz** [comment]
@hynek I agree with @vincent-stater, full module path is the name of module, - the second item of tuple returning by `_find_first_app_frame_and_name`.

Also I guess `CallsiteParameter.QUAL_NAME` is missing inside `CallsiteParameterAdder._record_attribute_map`.

**Comment by hynek** [comment]
> Also I guess `CallsiteParameter.QUAL_NAME` is missing inside `CallsiteParameterAdder._record_attribute_map`.

That's because I don't think there is a qual name on standard library's records?

https://docs.python.org/3.14/library/logging.html#logrecord-attributes

As for the rest, sorry for closing prematurely and the delay. I hope #812 fixes it for good?

**Comment by decaz** [comment]
@hynek thank you! Left some [comment](https://github.com/hynek/structlog/pull/812#discussion_r3266699694) there.
